Sources/Sentry/SentryClient.m

```diff
@@ -638,6 +642,23 @@ - (BOOL)isOOM:(SentryEvent *)event isCrashEvent:(BOOL)isCrashEvent
         [exception.mechanism.type isEqualToString:SentryOutOfMemoryMechanismType];
 }
 
+- (void)storeFreeMemoryToDeviceContext:(SentryEvent *)event
+{
+    if (nil == event.context || event.context.count == 0 || nil == event.context[@"device"]) {
+        return;
+    }
+
+    NSMutableDictionary *context = [[NSMutableDictionary alloc] initWithDictionary:event.context];
+    NSMutableDictionary *device =
+        [[NSMutableDictionary alloc] initWithDictionary:context[@"device"]];
+
+    NSDictionary *systemInfo = [[SentryCrashWrapper sharedInstance] systemInfo];
```

Comment:
By switching to `SentryCrash.sharedInstance.systemInfo` like @brustolin suggested, we get the uncached value and adding the log indeed shows it does change. Do you think that is an acceptable fix, or do you want to call the freeMemory method like you suggested?

Reply:
Really `SentryCrash.sharedInstance.systemInfo` is uncached? It didn't work for me though.

Reply:
My logs were definitely showing different values. With you they always shows the same?

Reply:
`SentryCrash.sharedInstance.systemInfo` is uncached
`SentryCrashWrapper.sharedInstance.systemInfo` is cached

Reply:
Yeah that's what I am seeing as well

Reply:
`SentryCrash.sharedInstance.systemInfo` has a lit bit of overhead because all of the information retrieved. Wondering if just exposing the `freeMemory` function is not a better solution.

Reply:
Alrighty, done. Confirmed that it works by sending multiple events and errors, and seeing the value actually change. Now, how would we unit test this?